src/components/search-paneset/search-paneset.js

```diff
@@ -37,17 +38,34 @@ export default class SearchPaneset extends React.Component {
   };
 
   state = {
-    hideFilters: this.props.hideFilters
+    hideFilters: this.props.hideFilters,
+    closePane: false
   };
 
-  componentWillReceiveProps({ hideFilters, resultsType }) {
+  componentWillReceiveProps({ hideFilters, resultsType, searchString }) {
     let isSameSearchType = resultsType === this.props.resultsType;
+    let isSameSearchString = searchString === this.props.searchString;
+
+    if (!isSameSearchString) {
+      this.setState({ closePane: true });
+    }
 
     if (isSameSearchType && hideFilters !== this.state.hideFilters) {
       this.setState({ hideFilters });
     }
   }
 
+  componentDidUpdate() {
+    this.shouldClosePane();
+  }
+
+  shouldClosePane = () => {
+    if (this.state.closePane) {
+      this.closePreview();
```

Comment:
I'm curious about the browser history at this point. If I understand correctly the flow is as follows: a new search happens and the URL is updated via a push, this component receives a new search string and calls this method, `closePreview` then pushes to the browser history thus closing the preview pane.

It seems like there would be two new entries in the browser history. So if you click the back button after a new search, we will get the same search but with the preview pane visible. If we click back again, then we get the old search with the same preview pane visible.

Also, what about when there isn't a preview pane open? If we create a new search without clicking on anything to preview I think those two browser history entries would still be there. Essentially requiring you to click back twice to actually go back to the old search.

I could be completely wrong about this, but that's what seems like is happening following through the code. Could we verify that this isn't the case?

Reply:
Great insight here Wil. I will check out all this areas that you pointed out to inspect what is happening.